deno/wipe_appwrite_bucket/src/mod.ts

```diff
@@ -0,0 +1,54 @@
+import { sdk } from "./deps.ts";
+
+export default async function(req: any, res: any) {
+  if(
+    !req.variables['APPWRITE_FUNCTION_ENDPOINT'] || 
+    !req.variables['APPWRITE_FUNCTION_PROJECT_ID'] || 
+    !req.variables['APPWRITE_FUNCTION_API_KEY'] ||
+    !req.payload
+  ) {
+    return res.json({
+      success: false,
+      message: "Missing required environment variables or payload.",
+    });
+  }
+  let bucketId: string = '';
+  try {
+    const payload = JSON.parse(req.payload);
+    bucketId = payload.bucketId;
+  } catch(err) {
+    return res.json({
+      success: false,
+      message: "Payload is invalid.",
+    });
+  }
+  const client = new sdk.Client();
+  const storage = new sdk.Storage(client);
+  
+  client
+    .setEndpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])
+    .setProject(req.variables['APPWRITE_FUNCTION_PROJECT_ID'])
+    .setKey(req.variables['APPWRITE_FUNCTION_API_KEY'])
+  ;
+
+  try {
+    const limit = 25;
+    let offset = 0;
+    let listFiles;
+    do {
+      listFiles = await storage.listFiles(bucketId, [sdk.Query.limit(limit), sdk.Query.offset(offset)]);
+      for(const file of listFiles.files) {
+        await storage.deleteFile(bucketId, file.$id);
+      }
+      offset += limit;
```

Comment:
Because you're deleting while fetching the files this offset can get a little out of sync. For example, if there were 26 files, after fetching the 1st 25 and deleting, when you fetch with offset = 25, you'll get 0 records.

Reply:
Oh, that's right. I am now using cursor pagination